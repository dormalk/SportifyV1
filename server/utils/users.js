class Users{
    constructor(){
      this.users = [];
    }
  
    addUser(socketId,uid,name){
      var user = {socketId,uid,name};
      this.users.push(user);
      return user;
    }
  
    removeUser(socketId){
      var user = this.getUser(socketId);
  
      if(user){
        this.users = this.users.filter((user) => user.socketId !== socketId);
      }
  
      return user;
    }
  
    getUser(socketId){
      return this.users.filter((user) => user.socketId === socketId)[0];
    }
  
    getUserList(){
      return this.users;
    }
  }
  
  module.exports = {Users};
  