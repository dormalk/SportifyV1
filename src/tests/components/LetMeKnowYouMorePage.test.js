import React from 'react';
import { shallow,mount } from 'enzyme';
import { 
    LetMeKnowYouMorePage,
    YourHobbies,
    YourAge,
    YourGender,
    CheckDetail,
} from '../../components/LetMeKnowYouMorePage';
import users from '../fixtures/users';
import database from '../../firebase/firebase';


test('should render LetMeKnowYouMore correctly', () => {
    const wrapper = shallow(<LetMeKnowYouMorePage/>);
    expect(wrapper).toMatchSnapshot();
});

test('should update fname on form load correctly', () => {
    const value = 'Dor';

    const wrapper = mount(<LetMeKnowYouMorePage user={users[0]}/>);
    wrapper.find('input').at(0).simulate('change',
        {target:{value}});
    wrapper.find('span').at(2).simulate('click');
    expect(wrapper.state('user').fname).toBe(value);
});


test('should update lname on form load correctly', () => {
    const value = 'Malka';
    const wrapper = mount(<LetMeKnowYouMorePage user={users[0]}/>);
    wrapper.find('input').at(1).simulate('change',
        {target:{value}});
    wrapper.find('span').at(4).simulate('click');
    expect(wrapper.state('user').lname).toBe(value);
});


test('should update fname/lname onClick button of CheckDetail', () => {
    const wrapper = shallow(<LetMeKnowYouMorePage user={users[0]}/>);
    expect(wrapper.state('user').fname).toBe(users[0].fname);
    expect(wrapper.state('user').lname).toBe(users[0].lname);
});


test('should render YourHobbies correctly', () => {
    const onHobbieEntered = jest.fn();
    const onHobbieRemoved = jest.fn();
    const onHobbieChanged = jest.fn();
    const onHobbieChangedFromSuggest = jest.fn();
    const suggestHobbie = jest.fn();
    const Hobbies = ['כדוררשת','כדורעף','כדורסל','כדורגל'];
    const currHobbie = [''];
    const wrapper = shallow(
        <YourHobbies 
            currHobbie={currHobbie}
            Hobbies={Hobbies}
            onHobbieEntered={onHobbieEntered}
            onHobbieRemoved={onHobbieRemoved}
            onHobbieChanged={onHobbieChanged}
            onHobbieChangedFromSuggest={onHobbieChangedFromSuggest}
            suggestHobbie={suggestHobbie}
            />
        );
    expect(wrapper).toMatchSnapshot();
});

test('should render YourAge correctly', () => {
    const wrapper = shallow(<YourAge/>);
    expect(wrapper).toMatchSnapshot();
});


test('should change detail property of state', () => {
    const gender = 'זכר';
    const age = 28;
    const hobbie = 'כדורגל';
    const moto = 'belive';

    const startEditUser = jest.fn();
    const wrapper = mount(<LetMeKnowYouMorePage user={users[0]} startEditUser={startEditUser}/>);

    wrapper.find('button').at(0).simulate('click');
    wrapper.find('input[type="radio"]').at(0).simulate('change',{target:{value:gender}});
    expect(wrapper.state('user').detail.gender).toBe(gender);

    wrapper.find('input[type="number"]').at(0).simulate('change',{target:{value:age}});
    expect(wrapper.state('user').detail.age).toBe(age);

    wrapper.find('input[type="text"]').at(0).simulate('change',{target:{value:hobbie}});
    wrapper.find('.suggestion').simulate('click',{target:{hobbie}});
    expect(wrapper.state('user').detail.hobbies[0]).toBe(hobbie);

    wrapper.find('.curr_hobbie button').simulate('click',{target:{hobbie}});
    expect(wrapper.state('user').detail.hobbies.length).toBe(0);

    wrapper.find('input[type="text"]').at(0).simulate('change',{target:{value:hobbie}});
    wrapper.find('.suggestion').simulate('click',{target:{hobbie}});
    wrapper.find('input[type="text"]').at(1).simulate('change',{target:{value:moto}});
    expect(wrapper.state('user').detail.moto).toBe(moto);

    wrapper.find('.let_me_know_you_more___finish_button').simulate('click',{preventDefault() {}});
    expect(startEditUser).toHaveBeenCalled();
});

test('should render YourGender correctly', () => {
    const wrapper = shallow(<YourGender/>);
    expect(wrapper).toMatchSnapshot();
});


test('should render CheckDetail correctly', () => {
    const wrapper = shallow(<CheckDetail/>);
    expect(wrapper).toMatchSnapshot();
});