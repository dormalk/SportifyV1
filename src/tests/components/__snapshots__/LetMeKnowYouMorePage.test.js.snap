// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CheckDetail correctly 1`] = `
<div
  className="check_detail"
>
  <h2>
    האם הפרטים נכונים?
  </h2>
  <EditabelInput
    label="שם פרטי"
  />
  <EditabelInput
    label="שם משפחה"
  />
  <button
    className="confirm_detail"
  >
    אישור
  </button>
</div>
`;

exports[`should render LetMeKnowYouMore correctly 1`] = `
<div
  className="content-container"
>
  <div
    className="let_me_know_you_more"
  >
    <CSSTransitionGroup
      transitionAppear={false}
      transitionEnter={true}
      transitionEnterTimeout={500}
      transitionLeave={false}
      transitionName="exm"
    >
      <div>
        <CheckDetail
          onChangeFname={[Function]}
          onChangeLname={[Function]}
          onContinueButton={[Function]}
        />
      </div>
    </CSSTransitionGroup>
  </div>
</div>
`;

exports[`should render YourAge correctly 1`] = `
<div
  className="your_age"
>
  <h2>
    גילך
  </h2>
  <input
    max="99"
    min="16"
    placeholder="18"
    type="number"
  />
</div>
`;

exports[`should render YourGender correctly 1`] = `
<div
  className="your_gender"
>
  <h2>
    מינך
  </h2>
  <div
    className="your_gender___option"
  >
    <label>
      <input
        name="gender"
        type="radio"
        value="זכר"
      />
      <span
        className="radiomark"
      >
        <span
          className="radiomark_checked"
        />
      </span>
      <span
        className="your_gender___label"
      >
        זכר
      </span>
    </label>
    <label>
      <input
        name="gender"
        type="radio"
        value="נקבה"
      />
      <span
        className="radiomark"
      >
        <span
          className="radiomark_checked"
        />
      </span>
      <span
        className="your_gender___label"
      >
        נקבה
      </span>
    </label>
    <label>
      <input
        name="gender"
        type="radio"
        value="לא מעוניין למסור"
      />
      <span
        className="radiomark"
      >
        <span
          className="radiomark_checked"
        />
      </span>
      <span
        className="your_gender___label"
      >
        לא מעוניין למסור
      </span>
    </label>
  </div>
</div>
`;

exports[`should render YourHobbies correctly 1`] = `
<div
  className="your__hobbies"
>
  <h2>
    תאר/י את עיסוקי הספורט שלך
  </h2>
  <input
    onChange={[Function]}
    onKeyPress={[Function]}
    type="text"
    value={
      Array [
        "",
      ]
    }
  />
  <div
    className="curr_hobbie__list"
  >
    <div
      className="curr_hobbie"
    >
      כדוררשת
      <button
        onClick={[Function]}
      >
        x
      </button>
    </div>
    <div
      className="curr_hobbie"
    >
      כדורעף
      <button
        onClick={[Function]}
      >
        x
      </button>
    </div>
    <div
      className="curr_hobbie"
    >
      כדורסל
      <button
        onClick={[Function]}
      >
        x
      </button>
    </div>
    <div
      className="curr_hobbie"
    >
      כדורגל
      <button
        onClick={[Function]}
      >
        x
      </button>
    </div>
  </div>
  <div
    className="suggestion__list"
  />
</div>
`;
